### PR TITLE
http: adding a default idle timeout of 1 hour

### PR DESCRIPTION
**Pull request by alyssawilk**
Adding a default 1 hour idle timeout to both upstream and downstream HTTP connections.  Having none set is a recipe for connection leaks.
Note this only applies for idle connections (no streams exist on the connection and none have been created or destroyed for a full hour)

Risk Level: High (changes behavior for many Envoy users)
Testing: new unit tests
Docs Changes: inline
Release Notes: included

I think this must also include an envoy-announce email for those who want connections doing absolutely nothing for longer than 1 hour.
@envoyproxy/senior-maintainers 
@jmarantz

### CODE REVIEW

**Comment by jmarantz** [comment]
qq: does this cover both client-proxy idle timeouts, and also proxy-upstream idle timeouts?

**Comment by alyssawilk** [comment]
@jmarantz both - was in relnotes but added to PR description too

File: docs/root/intro/version_history.rst [hunk 1]
```diff
@@ -32,6 +32,7 @@ Version history
 * grpc-json: added support for :ref:`ignoring unknown query parameters<envoy_api_field_config.filter.http.transcoder.v2.GrpcJsonTranscoder.ignore_unknown_query_parameters>`.
 * grpc-json: added support for :ref:`the grpc-status-details-bin header<envoy_api_field_config.filter.http.transcoder.v2.GrpcJsonTranscoder.convert_grpc_status>`.
 * header to metadata: added :ref:`PROTOBUF_VALUE <envoy_api_enum_value_config.filter.http.header_to_metadata.v2.Config.ValueType.PROTOBUF_VALUE>` and :ref:`ValueEncode <envoy_api_enum_config.filter.http.header_to_metadata.v2.Config.ValueEncode>` to support protobuf Value and Base64 encoding.
+* http: added a default 1 hour idle timeout to upstream and downstream connections. HTTP connections with no stream and no activity will be closed after 1 hour unless the default is overridden.
```
Comment (mattklein123) [inline docs/root/intro/version_history.rst]: sorry, forgot this before but can you ref link to the relevant idle timeout and/or protocol fields in the HCM/cluster so people know how to set this?

File: docs/root/intro/version_history.rst [hunk 2]
```diff
@@ -32,6 +32,7 @@ Version history
...
 * header to metadata: added :ref:`PROTOBUF_VALUE <envoy_api_enum_value_config.filter.http.header_to_metadata.v2.Config.ValueType.PROTOBUF_VALUE>` and :ref:`ValueEncode <envoy_api_enum_config.filter.http.header_to_metadata.v2.Config.ValueEncode>` to support protobuf Value and Base64 encoding.
+* http: added a default one hour idle timeout to upstream and downstream connections. HTTP connections with no stream and no activity will be closed after one hour unless the default idle_timeout overridden. To disable upstream idle timeouts, set the `idle_timeout <envoy_api_field_core.HttpProtocolOptions.idle_timeout>` to zero in Cluster :ref:`http_protocol_options<envoy_api_field_Cluster.common_http_protocol_options>`. To disable downstream idle timeouts, either set `idle_timeout <envoy_api_field_core.HttpProtocolOptions.idle_timeout>` to zero in the HttpConnectionManager :ref:`common_http_protocol_options <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.common_http_protocol_options>` or set the deprecated :ref:`connection manager <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.idle_timeout>` field to zero.
```
Comment (mattklein123) [inline docs/root/intro/version_history.rst]: You are missing some `:ref:` prefixes in here (check the rendered docs). Sorry for the runaround. :(

Comment (alyssawilk) [reply docs/root/intro/version_history.rst]: No, totally my bad.  I'm terrible at markup and thought docs/build.sh caught more than it does :-P

**Pull request merged**